src/core/xref.js

```diff
@@ -202,6 +171,52 @@ function getTermFromElement(elem) {
   return term === "the-empty-string" ? "" : term;
 }
 
+/**
+ * Get spec context as a fallback chain, where each level (sub-array) represents
+ * decreasing priority.
+ * @param {HTMLElement} elem
+ */
+function getSpecContext(elem) {
+  /** @type {string[][]} */
+  const specs = [];
+
+  /** @type {HTMLElement} */
+  let dataciteElem = elem.closest("[data-cite]");
+
+  // If element itself contains data-cite, we don't take inline context into
+  // account. The inline bibref context has highest priority, if available.
+  if (dataciteElem !== elem) {
+    const closestSection = elem.closest("section");
+    /** @type {Iterable<HTMLElement>} */
+    const bibrefs = closestSection
+      ? closestSection.querySelectorAll("a.bibref")
+      : [];
+    const inlineRefs = [...bibrefs].map(el => el.textContent.toLowerCase());
+    const uniqueInlineRefs = [...new Set(inlineRefs)].sort();
+    if (uniqueInlineRefs.length) {
+      specs.push(uniqueInlineRefs);
+    }
+  }
+
+  // Traverse up towards the root element, adding levels of lower priority specs
+  while (dataciteElem) {
+    const cite = dataciteElem.dataset.cite.toLowerCase().replace(/[!?]/g, "");
+    const cites = cite.split(/\s+/).filter(s => s);
+    // Optimization: if we already have a spec in a higher priority level of
+    // fallback chain, skip it from this level
+    const uniqueCites = [...new Set(cites)].filter(
+      spec => !(specs[specs.length - 1] || []).includes(spec)
+    );
+    if (uniqueCites.length) {
+      specs.push(uniqueCites.sort());
+    }
```

Comment:
I meant a separate function to make the array items unique and thought it could be done within this PR 👀. For unrelated refactors I prefer a separate PR. Sorry to confuse you 😔

Reply:
Oops :man_facepalming: Let me know if the current update is what you mean